applets/wncklet/showdesktop.c

```diff
@@ -361,14 +359,20 @@ static void show_desktop_applet_realized(MatePanelApplet* applet, gpointer data)
 
 	sdd = (ShowDesktopData*) data;
 
-	if (sdd->wnck_screen != NULL)
-		g_signal_handlers_disconnect_by_func(sdd->wnck_screen, show_desktop_changed_callback, sdd);
-
 	if (sdd->icon_theme != NULL)
 		g_signal_handlers_disconnect_by_func(sdd->icon_theme, theme_changed_callback, sdd);
 
 	screen = gtk_widget_get_screen(sdd->applet);
-	sdd->wnck_screen = wnck_screen_get(gdk_x11_screen_get_screen_number (screen));
+
+	if (sdd->wnck_screen != NULL)
+		g_signal_handlers_disconnect_by_func(sdd->wnck_screen, show_desktop_changed_callback, sdd);
+
+	sdd->wnck_screen = NULL;
+
+#ifdef HAVE_X11
+	if (GDK_IS_X11_DISPLAY (gdk_display_get_default ()))
+		sdd->wnck_screen = wnck_screen_get(gdk_x11_screen_get_screen_number (screen));
+#endif // HAVE_X11
```

Comment:
Do not add C ++ style comments (Single-line comments) like the one in the previous line.
Please, use C style comments on C code.

Further information is available https://en.wikibooks.org/wiki/C%2B%2B_Programming/Code/Style_Conventions/Comments

Reply:
Is there a consensus among the MATE developers that single-line comments are bad? Why?

I'd like to point out I was not the first one to start using single-line comments in mate-panel. Also the page you link includes

> The use of one-line C-style comments should be avoided as they are considered outdated.

I get the consistency argument, but since some instances of both are already there, I think that's weaker. Perhaps we should run a regex to make all single-line comments C++-style? That should certainly be another PR though.